src/OpenTelemetry/Metrics/MetricPoint.cs

```diff
@@ -69,15 +74,65 @@ public struct MetricPoint
         /// </summary>
         public ReadOnlyTagCollection Tags { get; }
 
-        public DateTimeOffset StartTime { get; internal set; }
+        internal MetricPointStatus MetricPointStatus { get; private set; }
 
-        public DateTimeOffset EndTime { get; internal set; }
+        public DateTimeOffset GetStartTime()
+        {
+            return this.StartTime;
+        }
 
-        public long LongValue { get; internal set; }
+        public DateTimeOffset GetEndTime()
+        {
+            return this.EndTime;
+        }
 
-        public double DoubleValue { get; internal set; }
+        public long GetLongSum()
+        {
+            if (this.aggType == AggregationType.LongSumIncomingDelta || this.aggType == AggregationType.LongSumIncomingCumulative)
+            {
+                return this.longValue;
+            }
+            else
+            {
+                throw new NotSupportedException($"{nameof(this.GetLongSum)} is not supported for this metric type.");
+            }
+        }
 
-        internal MetricPointStatus MetricPointStatus { get; private set; }
+        public double GetDoubleSum()
+        {
+            if (this.aggType == AggregationType.DoubleSumIncomingDelta || this.aggType == AggregationType.DoubleSumIncomingCumulative)
+            {
+                return this.doubleValue;
+            }
+            else
+            {
+                throw new NotSupportedException($"{nameof(this.GetDoubleSum)} is not supported for this metric type.");
+            }
+        }
+
+        public long GetLastLongValue()
```

Comment:
```suggestion
        public long GetLastValueLong()
```

Reply:
Is "last value" something meaningful from the spec? If not something like `GetGaugeLongValue` might be more clear.

Reply:
https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#last-value-aggregation

Reply:
"Last Value" https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#last-value-aggregation

"Gauge" https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/datamodel.md#gauge

Reply:
Basically, Gauge in OpenTelemetry is a stream of metric points, and each point is a combination of **{last value, timestamps, attributes, exemplars}**.

Reply:
We do have some helper methods with "Histogram" in the name (at least currently), what if we added a type in all the names? Something like...

`GetCounterSumAsLong, GetCounterSumAsDouble, GetGaugeLastValueAsLong, GetGaugeLastValueAsDouble`

Reply:
^ I like this, except the "As" in the name which indicates we are doing some cast..
`GetCounterSumLong` and friends, sounds better to me.